src/mito2/src/memtable/time_partition.rs

```diff
@@ -229,23 +220,7 @@ impl TimePartitions {
         next_memtable_id: MemtableId,
         part_duration: Option<Duration>,
     ) -> Self {
-        let mut inner = PartitionsInner::new(next_memtable_id);
-        if part_duration.is_none() {
-            // If `part_duration` is None, then we create a partition with `None` time
-            // range so we will write all rows to that partition.
-            let memtable = builder.build(inner.alloc_memtable_id(), &metadata);
-            debug!(
-                "Creates a time partition for all timestamps, region: {}, memtable_id: {}",
-                metadata.region_id,
-                memtable.id(),
-            );
-            let part = TimePartition {
-                memtable,
-                time_range: None,
-            };
-            inner.parts.push(part);
-        }
-
+        let inner = PartitionsInner::new(next_memtable_id);
```

Comment:
The following comment is outdated. The time_range is not None.
https://github.com/GreptimeTeam/greptimedb/blob/b4d00fb49957030c0577a84cffbf8a7d0b1366ac/src/mito2/src/memtable/time_partition.rs#L211-L215

Maybe we can always initialize the duration to 1 day so we don't need to wrap `TimePartitions::part_duration` into `Option`.

Reply:
fixed in 47abc221669a550fc48b03ba01ff0bb995319ccd